### ISSUE TITLE
Change Request: Flat Rule Tester support filename for non-JavaScript files

### ISSUE THREAD

**Issue by DMartens** [eslint/eslint#17577]
### ESLint version

8.49.0

### What problem do you want to solve?

The Flat Rule Tester currently reports an error when setting the `filename` of a test case to a non-JavaScript file (e.g. `file.ts`):  
`No matching configuration found for file.ts`.  
As I understand it, this is because files ending with `.js` are always matched in the flat config system even if there is no `files` property.

### What do you think is the correct solution?

Translate the `filename` property to `files` in the [config passed to the linter](https://github.com/eslint/eslint/blob/aa1b657a9febcd03e9298c03ae2888762795e322/lib/rule-tester/flat-rule-tester.js#L587C21-L587C21):
```javascript
...
const itemConfig = { ...item }
if (item.filename) {
	itemConfig.files = [item.filename]
}
...
```
This ensures that the generated config matches the file.

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

This problem can also be circumvented by providing a `baseConfig` but this solution is more ergonomical and this was also not required when using the "legacy" rule tester, so this may be seen as a bug.

**Comment by mdjermanovic** [comment]
> No matching configuration found for file.ts

I can reproduce this. `filename` must match at least one `files` pattern that doesn't end with `/*` or `/**`. Otherwise, it is considered that there is no configuration for that file, meaning that the file shouldn't be linted. `.js` files match a pattern in the default config, so they're always included:

https://github.com/eslint/eslint/blob/aa1b657a9febcd03e9298c03ae2888762795e322/lib/config/default-config.js#L56-L59

> Translate the `filename` property to `files` in the [config passed to the linter](https://github.com/eslint/eslint/blob/aa1b657a9febcd03e9298c03ae2888762795e322/lib/rule-tester/flat-rule-tester.js#L587C21-L587C21):

I support this proposal. I actually think the current behavior can be considered a bug. @nzakas what do you think?

We could add `files: [item.filename]` to some of the config objects FlatRuleTester already creates or add another config object (`configs.push({ files: [item.filename] })`).

**Comment by mdjermanovic** [comment]
> I think, ideally, we should try to mimic the behavior of `RuleTester` where `context.filename` is set but otherwise the `filename` option has no effect.
> 
> While we could manually add `files: [item.filename]`, it might make more sense to add `"**/*.*"`?

Makes sense to me 👍

**Comment by nzakas** [comment]
@DMartens do you want to submit a PR for this?

**Comment by DMartens** [comment]
Yes, already working on one.

**Comment by mdjermanovic** [comment]
Actually, a problem with `"**/*.*"` would be that it won't work for filenames that don't have extension (i.e., don't have a `.`). For example, `filename: "somefile"`.
